File: 1/main.jl
function getFuel(mass)
    total = 0
    for i in mass
       total = max(floor(parse(Int, i)/3) - 2, 0) + total
    end
    return Int(total)
end

function getFuelOfFuel(mass)
    total = 0
    for i in mass
       j = parse(Int, i)
       while j > 0
           total = max(floor(j/3) - 2, 0) + total
           j = max(floor(j/3) - 2, 0)
        end
    end
    return Int(total)
end

function main()
    f = open("input")
    mass = readlines(f)
    totalFuel = getFuel(mass)
    totalFuelOfFuel = getFuelOfFuel(mass)
    println("total fuel: ", totalFuel)
    println("total fuel of fuel: ", totalFuelOfFuel)
end

main()

File: 1/main.py
def main():
    mass = [int(line.rstrip('\n')) for line in open("input")]
    totalFuel = getFuel(mass)
    totalFuelOfFuel = getFuelOfFuel(mass)
    print("Total fuel: " + str(totalFuel))
    print("Total fuel of fuel: " + str(totalFuelOfFuel))

def getFuel(mass):
    total = 0
    for i in mass:
        total = total + max((i // 3 - 2), 0)
    return total

def getFuelOfFuel(mass):
    total = 0
    for i in mass:
        j = i
        while j > 0:
            total = total + max((j // 3 - 2), 0)
            j = max((j // 3 - 2), 0)
    return total


if __name__ == '__main__':
    main()
